### PR TITLE
add api directory

### PR DESCRIPTION
**Pull request by zain510**

### CODE REVIEW

File: lib/json_web_token.rb [hunk 1]
```diff
@@ -1,4 +1,3 @@
-
 class JsonWebToken
   SECRET_KEY = Rails.application.secrets.secret_key_base. to_s
```
Comment (RakeshTejra) [inline lib/json_web_token.rb]: Remove space before `.to_s`

File: app/controllers/api/v1/candidates/candidates_controller.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+class Api::V1::Candidates::CandidatesController < ApplicationController
+
+  def sign_up
```
Comment (RakeshTejra) [inline app/controllers/api/v1/candidates/candidates_controller.rb]: method name should be `create`, that is more relevant to REST style.

File: app/controllers/api/v1/candidates/session_controller.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class Api::V1::Candidates::SessionController < ApplicationController
+
+  def login
```
Comment (RakeshTejra) [inline app/controllers/api/v1/candidates/session_controller.rb]: again should be `create` action

File: app/controllers/api/v1/job_categories_controller.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class Api::V1::JobCategoriesController < ApplicationController
+  before_action :authorize_request
+  
+  def index
+    job_categories = JobCategory.all
+  	render json: job_categories
```
Comment (RakeshTejra) [inline app/controllers/api/v1/job_categories_controller.rb]: not indent

File: app/controllers/api/v1/job_categories_controller.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+class Api::V1::JobCategoriesController < ApplicationController
+  before_action :authorize_request
```
Comment (RakeshTejra) [inline app/controllers/api/v1/job_categories_controller.rb]: Instead adding this filter to every controller we can move it to parent class

**Review by zrizvi1 — COMMENTED** [review]
change

**Comment by zrizvi1** [comment]
change

**Pull request merged**